src/csharp/Microsoft.Spark.E2ETest/IpcTests/Sql/DataFrameTests.cs

```diff
@@ -266,36 +269,18 @@ public void TestGroupedMapUdf()
             }
         }
 
-        private static RecordBatch CountCharacters(RecordBatch records)
+        private static FxDataFrame CountCharacters(FxDataFrame dataFrame)
         {
-            int stringFieldIndex = records.Schema.GetFieldIndex("name");
-            StringArray stringValues = records.Column(stringFieldIndex) as StringArray;
-
             int characterCount = 0;
-
-            for (int i = 0; i < stringValues.Length; ++i)
+            PrimitiveColumn<int> characterCountColumn = new PrimitiveColumn<int>("name" + "_CharCount", dataFrame.RowCount);
+            for (long i = 0; i < dataFrame.RowCount; i++)
             {
-                string current = stringValues.GetString(i);
-                characterCount += current.Length;
+                characterCount += ((string)dataFrame["name"][i]).Length;
+                characterCountColumn[i] = characterCount;
             }
-
-            int groupFieldIndex = records.Schema.GetFieldIndex("age");
-            Field groupField = records.Schema.GetFieldByIndex(groupFieldIndex);
-
-            // Return 1 record, if we were given any. 0, otherwise.

```

Comment:
Similar comment as the Examples VectorUdfs.cs file. (since this is duplicated code)